packages/contracts-bedrock/src/Safe/VetoerSet/OwnerGuard.sol

```diff
@@ -0,0 +1,134 @@
+
+        // Set the initial `maxOwnerCount`, to the greater between `INITIAL_MAX_OWNER_COUNT` and the current owner
+        // count.
+        maxOwnerCount = uint8(FixedPointMathLib.max(INITIAL_MAX_OWNER_COUNT, ownerCount));
+    }
+
+    /// @notice Inherited hook from `BaseGuard` that is run right before the transaction is executed
+    ///         by the Safe Account when `execTransaction` is called.
+    function checkTransaction(
+        address,
+        uint256,
+        bytes memory,
+        Enum.Operation,
+        uint256,
+        uint256,
+        uint256,
+        address,
+        address payable,
+        bytes memory,
+        address
+    )
+        external
+    { }
+
+    /// @notice Inherited hook from `BaseGuard` that is run right after the transaction has been executed
+    ///         by the Safe Account when `execTransaction` is called.
+    function checkAfterExecution(bytes32, bool) external view {
+        // Ensure the length of the new set of owners is not above `maxOwnerCount`, and get the corresponding threshold.
+        uint256 expectedThreshold = checkNewOwnerCount(safe.getOwners().length);
+
+        // Ensure the Safe Account threshold always stays in sync with the 66% one.
+        uint256 threshold = safe.getThreshold();
+        if (threshold != expectedThreshold) {
+            revert InvalidSafeAccountThreshold(threshold, expectedThreshold);
+        }
+    }
+
+    /// @notice Update the maximum number of owners.
+    /// @dev Reverts if not called by the Safe Account.
+    /// @param newMaxOwnerCount The new possible `maxOwnerCount` of owners.
+    function updateMaxOwnerCount(uint8 newMaxOwnerCount) external {
+        // Ensure only the Safe Account can call this function.
+        if (msg.sender != address(safe)) {
+            revert SenderIsNotSafeAccount(msg.sender);
+        }
+
+        // Ensure the given `newMaxOwnerCount` is not bellow the current number of owners.
+        uint256 ownerCount = safe.getOwners().length;
+        if (newMaxOwnerCount < ownerCount) {
+            revert MaxOwnerCountTooLow(newMaxOwnerCount, ownerCount);
+        }
+
+        // Update the new `maxOwnerCount`.
+        maxOwnerCount = newMaxOwnerCount;
+    }
+
+    /// @notice Checks if the given `newOwnerCount` of owners is allowed and returns the corresponding 66% threshold.
+    /// @dev Reverts if `newOwnerCount` is above `maxOwnerCount`.
+    /// @param newOwnerCount The owners count to check.
+    /// @return threshold The corresponding 66% threshold for `newOwnerCount` owners.
+    function checkNewOwnerCount(uint256 newOwnerCount) public view returns (uint256 threshold) {
+        // Ensure we don't exceed the maximum number of allowed owners.
+        if (newOwnerCount > maxOwnerCount) {
+            revert OwnerCountTooHigh(newOwnerCount, maxOwnerCount);
+        }
+
+        // Compute the corresponding ceil (66%) threshold of owners.
+        threshold = (newOwnerCount * 66 + 99) / 100;
```

Comment:
do we want this to be configurable?

Reply:
To my understanding this supermajority (>= 2/3) should never change.